crates/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -206,12 +206,9 @@ struct Branch {
 }
 
 fn flatten_function_cfg(function: &mut Function) {
-    // TODO This pass will run forever on a brillig function.
-    // TODO In particular, analyze will check if the predecessors
-    // TODO have been processed and push the block to the back of the queue
-    // TODO This loops forever, if the predecessors are not then processed
-    // TODO Because it will visit the same block again, pop it out of the queue
-    // TODO then back into the queue again.
+    // This pass may run forever on a brillig function.
+    // Analyze will check if the predecessors have been processed and push the block to the back of
+    // the queue. This loops forever if there are still any loops present in the program.
     if let crate::ssa::ir::function::RuntimeType::Brillig = function.runtime() {
         return;
```

Comment:
Should we still not be flattening for Brillig funcs if we are applying optimizations now? I don't see any restrictions for brillig funcs in the loop unrolling pass?

Reply:
As the comment mentions, flattening will loop forever for any brillig functions which still have unbounded loops in them. There is an exemption for brillig functions in loop unrolling - if we ever fail to unroll a loop it is reported as an error for acir functions or it is just skipped over for brillig functions.

Reply:
Ah yes thank you for clarifying